source/write/bulk-write.txt

```diff
@@ -0,0 +1,343 @@
+``BulkWriteError`` and does not perform any further operations.
+``BulkWriteError`` provides a ``details`` item that includes the
+operation that failed, and details about the exception.
+
+.. note::
+
+   When the driver runs a bulk operation, it uses the write concern of the
+   target collection. The driver reports all write concern errors after
+   attempting all operations, regardless of execution order.
+
+Customize Bulk Write Operation
+------------------------------
+
+The ``bulkWrite()`` method optionally accepts a parameter which
+specifies options you can use to configure the bulk write
+operation. If you don't specify any options, the driver performs the
+bulk operation with default settings.
+
+The following table describes the setter methods that you can use to
+configure a ``BulkWriteOptions`` instance:
+
+.. list-table::
+   :widths: 30 70
+   :header-rows: 1
+
+   * - Method
+     - Description
+
+   * - ``ordered()``
+     - | If ``true``, the driver performs the write operations in the order
+         provided. If an error occurs, the remaining operations are not
+         attempted. 
+       |
+       | If ``false``, the driver performs the operations in an
+         arbitrary order and attempts to perform all operations.
+       | Defaults to ``true``.
+
+   * - ``bypassDocumentValidation()``
+     - | Specifies whether the update operation bypasses document validation. This lets you 
+         update documents that don't meet the schema validation requirements, if any 
+         exist. For more information about schema validation, see :manual:`Schema
+         Validation </core/schema-validation/#schema-validation>` in the MongoDB
+         Server manual.
+       | Defaults to ``false``.
+
+   * - ``comment()``
+     - | Sets a comment to attach to the operation.
+
+   * - ``let()``
+     - | Provides a map of parameter names and values to set top-level
+         variables for the operation. Values must be constant or closed
+         expressions that don't reference document fields.
+
+The following code creates options and uses the ``ordered(false)`` option to
+specify an unordered bulk write. Then, the example uses the
+``bulkWrite()`` method to perform a bulk operation:
+
+.. literalinclude:: /includes/write/bulk-write.scala
+   :start-after: start-bulk-write-unordered
+   :end-before: end-bulk-write-unordered
+   :language: scala
+   :copyable:
+   :dedent:
+
+If any of the write operations in an unordered bulk write fail, the {+driver-short+}
+reports the errors only after attempting all operations.
+
+.. note::
+
+   Unordered bulk operations do not guarantee an order of execution. The
+   order can differ from the way you list them to optimize the runtime.
+
+Return Value
+------------
+
+The ``bulkWrite()`` method returns a ``SingleObservable`` object. You can subscribe to
+this observable to access information from a ``BulkWriteResult`` instance by using the
```

Comment:
S: this wording is a little confusing, since it's unclear if the method returns a SingleObservable or a BulkWriteResult. From the API docs, it looks like it returns a SingleObservable that contains a BulkWriteResult. Maybe reword to: 

```suggestion
The ``bulkWrite()`` method returns a ``SingleObservable`` object that contains a ``BulkWriteResult``. You can access information from the ``BulkWriteResult`` instance by subscribing to the observable and using the
```